### ISSUE TITLE
motion guard might could use a cork/inhibit during ptz operations

### ISSUE THREAD

**Issue by goatzillax** [themactep/thingino-firmware#324]
Maybe PTZ can drop a cork or inhibit file in /tmp that the motion script will check for before firing, and then delete the cork when PTZ is finished?

**Comment by themactep** [comment]
good idea. @gtxaspec, if you add a pid file (say /run/motors.pid) to `motors` app then i'll make `motion` script to check for that file and bail out if it is set.

**Comment by themactep** [comment]
won't that run another copy of motors with all the overhead?

**Comment by gtxaspec** [comment]
it will, but it works effectively while keeping things lightweight. the app is tiny and exits once it completes, which eliminates us from adding more bloat in the daemon.  overhead is minimal and won't be noticed.

**Comment by goatzillax** [comment]
The app does look tiny; I wonder if there is potential for a race between the intention to move, motor, and the motion script finally getting back the status that the motors are now indeed busy.

Seems like the cork mechanism should be dropped before even calling motor so that the intention can be communicated to the motion script, in order, and as early as possible.

**Comment by goatzillax** [comment]
Hrm, I was playing around with wrapping motors with a script that drops a cork for motion look for.

Should motors block until the movement is completed, or would that not be in keeping with design?

**Comment by themactep** [comment]
implemented. please test tomorrow build and close the ticket if satisfied with the results.

**Comment by goatzillax** [comment]
This seems to work well enough to close the issue.

FYI I don't know if this is already fixed, but /sbin/motion is running jct get on /etc/prudynt.json for motion settings when it appears to need to look in /etc/motion.json instead.  Had to fix that to fully test the feature.
